File: projects/collatz/julia/collatz.jl

function col(i :: Int)
	length = 0
	num = i
	while i != 1
		if i % 2 == 0
			i ÷= 2
			if (i > num)
				num = i
			end
		else
			i = i * 3 + 1
			if (i > num)
				num = i
			end
		end
		length += 1
	end
	return num
end

function main()

	arg1 = ARGS[1]
	arg2 = ARGS[2]
	
	startStr = arg1[1:2] * arg1[4:5] * arg1[7:10]
	finishStr = arg2[1:2] * arg2[4:5] * arg2[7:10]
	start = parse(Int, startStr)
	finish = parse(Int, finishStr)
	num = 0
	for i in start : finish
		n = col(i)
		if (n > num)
			num = n	
		end	
	end

	println(num)
end

main()

File: projects/collatz/python/collatz.py
import sys

def col(i):
    num = i
    while i != 1:
        if i % 2 == 0:
            i //= 2
            if i > num:     
                num = i
        else:
            i = i * 3 + 1
            if i > num:
                num = i
    return num


def main():

    arg1 = sys.argv[1]
    arg2 = sys.argv[2]

    startStr = arg1.replace("/","")
    finishStr = arg2.replace("/","")
    start = int(startStr)
    finish = int(finishStr)
    
    

    num = 0
    for i in range(start, finish + 1):
        n = col(i)
        if n > num:
            num = n 
         
    print(num)


if __name__ == "__main__":
    main()

